x/metadata/keeper/keeper_test.go

```diff
@@ -460,6 +592,106 @@ func (s *KeeperTestSuite) TestValidateAllOwnersAreSigners() {
+		grantee               sdk.AccAddress
+		errorMsg              string
+	}{
+		// authz test cases
+		{
+			name:                  "Scope Spec with 2 owners - one signer - with grant - authz",
+			owners:                []string{s.user2, s.user3},
+			signers:               []string{s.user3},
+			msgTypeURL:            types.TypeURLMsgAddScopeOwnerRequest,
+			allowedAuthorizations: oneAllowedAuthorizations,
+			granter:               s.user2Addr,
+			grantee:               s.user3Addr,
+			errorMsg:              "",
+		},
+		{
+			name:                  "Scope Spec with 2 owners - one signer - no grant - authz - error",
+			owners:                []string{s.user2, s.user3},
+			signers:               []string{s.user2},
+			msgTypeURL:            types.TypeURLMsgWriteScopeRequest,
+			allowedAuthorizations: manyAllowedAuthorizations,
+			granter:               s.user2Addr,
+			grantee:               s.user3Addr,
+			errorMsg:              fmt.Sprintf("missing signature from existing owner %s; required for update", s.user3),
+		},
+		{
+			name:                  "Scope Spec with 3 owners - one signer with a special case message type - with grant - authz",
+			owners:                []string{s.user1, s.user2, s.user3},
+			signers:               []string{s.user1, s.user3}, // signer 3 is grantee of singer 2
+			msgTypeURL:            types.TypeURLMsgAddScopeDataAccessRequest,
+			allowedAuthorizations: manyAllowedAuthorizations,
+			granter:               s.user2Addr,
+			grantee:               s.user3Addr,
+			errorMsg:              "",
+		},
+		{
+			name:                  "Scope Spec with 3 owners - two signers with a special case message type - grant on parent of special case message type - authz",
+			owners:                []string{s.user1, s.user2, s.user3},
+			signers:               []string{s.user1, s.user3}, // signer 3 grantee of signer 2
+			msgTypeURL:            types.TypeURLMsgDeleteContractSpecFromScopeSpecRequest,
+			allowedAuthorizations: manyAllowedAuthorizations,
+			granter:               s.user2Addr,
+			grantee:               s.user3Addr,
+			errorMsg:              "",
+		},
+		{
+			name:                  "Scope Spec with 2 owners - one signer - no grant - authz - error",
+			owners:                []string{s.user2, s.user3},
+			signers:               []string{s.user3},
+			msgTypeURL:            types.TypeURLMsgDeleteRecordRequest,
+			allowedAuthorizations: manyAllowedAuthorizations,
+			granter:               nil,
+			grantee:               nil,
+			errorMsg:              fmt.Sprintf("missing signature from existing owner %s; required for update", s.user2),
+		},
+	}
+
+	now := s.ctx.BlockHeader().Time
+	s.Require().NotNil(now)
+
+	// Test cases
+	for _, tc := range tests {
+		s.T().Run(tc.name, func(t *testing.T) {
+			a := authz.NewCountAuthorization(tc.msgTypeURL, tc.allowedAuthorizations)
+			err := s.app.AuthzKeeper.SaveGrant(s.ctx, tc.grantee, tc.granter, a, now.Add(time.Hour))
+			require.NoError(t, err)
```

Comment:
Here too with the only-if `granter` and `grantee` aren't nil thing and require message thing.